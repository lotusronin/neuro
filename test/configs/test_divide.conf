conf = {
'src' : 'test_divide.nro',
'expected' : 'test_divide.expected'
}
