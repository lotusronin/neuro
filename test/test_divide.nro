extern fn putchar(c : int) : int;
extern fn puts(s :*u8) : int;

fn main() : u32 {
    var1 := 30;
    var2 := 5;
    puts("30 / 5 is...?");
    printi(var1/var2);
    putchar('\n');
    return 0;
}

fn printi(i : u32) : u32 {
    if(i < 0) {
        putchar('-');
        i = 0-i;
    }
    if(i < 10) {
        putchar('0'+i);
    } else {
        c := i/10;
        printi(c);
        n := i - c*10;
        putchar('0'+n);
    }
    return 0;
}
