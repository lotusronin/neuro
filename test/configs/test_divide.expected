30 / 5 is...?
6
